airflow/jobs.py

```diff
@@ -1140,6 +1138,7 @@ def _execute_task_instances(self,
 
                 open_slots -= 1
                 dag_id_to_possibly_running_task_count[dag_id] += 1
+        session.commit()
```

Comment:
This will have the consequence of preventing any tasks from being executed if there is an issue in e.g. the executor that throws an exception (like we saw with the CeleryExecutor at Airbnb). Probably ok though.